src/CommunityCommons/javasource/communitycommons/XPath.java

```diff
@@ -33,806 +33,829 @@
-
-	}
-
-	public static Class<?> getProxyClassForEntityName(String entityname)
-	{
-		{
-			String [] parts = entityname.split("\\.");
-			try
-			{
-				return Class.forName(parts[0].toLowerCase() + ".proxies." +  parts[1]);
-			}
-			catch (ClassNotFoundException e)
-			{
-				throw new RuntimeException("Cannot find class for entity: " + entityname + ": " + e.getMessage(), e);
-			}
-		}
-	}
-
-	public boolean deleteAll() throws CoreException
-	{
-		this.limit(1000);
-		List<IMendixObject> objs = allMendixObjects();
-		while (!objs.isEmpty()) {
-			if (!Core.delete(context, objs.toArray(new IMendixObject[objs.size()]))) 
-				return false; //TODO: throw?
-			
-			objs = allMendixObjects();
-		}
-		return true;
-	}
-
-	// Implement our own ESCAPE_XML because the original got deprecated and we originally translated only a very small
-	// subset of characters.
-	private static final CharSequenceTranslator ESCAPE_XML =
-		new AggregateTranslator(
-			new LookupTranslator(EntityArrays.BASIC_ESCAPE),
-			new LookupTranslator(EntityArrays.APOS_ESCAPE)
-		);
+public class XPath<T> {
+  /**
+   * Build in tokens, see:
+   * https://world.mendix.com/display/refguide3/XPath+Keywords+and+System+Variables
+   * 
+   */
+  public static final String CurrentUser = "[%CurrentUser%]";
+  public static final String CurrentObject = "[%CurrentObject%]";
+
+  public static final String CurrentDateTime = "[%CurrentDateTime%]";
+  public static final String BeginOfCurrentDay = "[%BeginOfCurrentDay%]";
+  public static final String EndOfCurrentDay = "[%EndOfCurrentDay%]";
+  public static final String BeginOfCurrentHour = "[%BeginOfCurrentHour%]";
+  public static final String EndOfCurrentHour = "[%EndOfCurrentHour%]";
+  public static final String BeginOfCurrentMinute = "[%BeginOfCurrentMinute%]";
+  public static final String EndOfCurrentMinute = "[%EndOfCurrentMinute%]";
+  public static final String BeginOfCurrentMonth = "[%BeginOfCurrentMonth%]";
+  public static final String EndOfCurrentMonth = "[%EndOfCurrentMonth%]";
+  public static final String BeginOfCurrentWeek = "[%BeginOfCurrentWeek%]";
+  public static final String EndOfCurrentWeek = "[%EndOfCurrentWeek%]";
+
+  public static final String DayLength = "[%DayLength%]";
+  public static final String HourLength = "[%HourLength%]";
+  public static final String MinuteLength = "[%MinuteLength%]";
+  public static final String SecondLength = "[%SecondLength%]";
+  public static final String WeekLength = "[%WeekLength%]";
+  public static final String YearLength = "[%YearLength%]";
+  public static final String ID = "id";
+  /** End builtin tokens */
+
+  private String entity;
+  private int offset = 0;
+  private int limit = -1;
+  // important, linked map, insertion order is relevant!
+  private LinkedHashMap<String, String> sorting = new LinkedHashMap<String, String>();
+  private LinkedList<String> closeStack = new LinkedList<String>();
+  private StringBuffer builder = new StringBuffer();
```

Comment:
As discussed in the standup, we may revert the commit that changes this StringBuilder to a StringBuffer.